dask/array/random.py

```diff
@@ -258,63 +552,80 @@ def choice(self, a, size=None, replace=True, p=None, chunks="auto"):
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def gamma(self, shape, scale=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("gamma", shape, scale, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "gamma", shape, scale, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def geometric(self, p, size=None, chunks="auto", **kwargs):
-        return self._wrap("geometric", p, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "geometric", p, size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def gumbel(self, loc=0.0, scale=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("gumbel", loc, scale, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "gumbel", loc, scale, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def hypergeometric(self, ngood, nbad, nsample, size=None, chunks="auto", **kwargs):
-        return self._wrap(
-            "hypergeometric", ngood, nbad, nsample, size=size, chunks=chunks, **kwargs
+        return _wrap_func(
+            self,
+            "hypergeometric",
+            ngood,
+            nbad,
+            nsample,
+            size=size,
+            chunks=chunks,
+            **kwargs,
         )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def laplace(self, loc=0.0, scale=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("laplace", loc, scale, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "laplace", loc, scale, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def logistic(self, loc=0.0, scale=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("logistic", loc, scale, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "logistic", loc, scale, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def lognormal(self, mean=0.0, sigma=1.0, size=None, chunks="auto", **kwargs):
-        return self._wrap("lognormal", mean, sigma, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(
+            self, "lognormal", mean, sigma, size=size, chunks=chunks, **kwargs
+        )
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def logseries(self, p, size=None, chunks="auto", **kwargs):
-        return self._wrap("logseries", p, size=size, chunks=chunks, **kwargs)
+        return _wrap_func(self, "logseries", p, size=size, chunks=chunks, **kwargs)
 
     @derived_from(np.random.RandomState, skipblocks=1)
     def multinomial(self, n, pvals, size=None, chunks="auto", **kwargs):
-        return self._wrap(
+        return _wrap_func(
+            self,
```

Comment:
I know you're not changing this here, but `kwargs` seems to be missing as well, just like in `Generator.multinomial`. GH won't allow me to suggest that edit, but perhaps we should add it. What do you think?

Reply:
good call. added with 3373f3e